hack/go-lint.sh

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-command -v golint > /dev/null || go get -u -modfile=go.tools.mod golang.org/x/lint/golint
+command -v golint > /dev/null || go install -modfile=go.tools.mod golang.org/x/lint/golint@latest
```

Comment:
```suggestion
command -v golint > /dev/null || go install golang.org/x/lint/golint@latest
```

Reply:
we can probably remove the go.tools.mod and sum as well